trk/src/Components/VideoGrid.js

```diff
@@ -75,24 +79,33 @@ const VideoGrid = () => {
   }, [selectedSetId, sets]);
 
   const renderVideoItem = ({ item, index }) => {
-    const isVisible = viewableItems.has(item);
+    // Extract video URL, accommodating both string URLs and object { url, role } formats
+    const videoUrl = typeof item === 'object' && item.url ? item.url : item;
+
+    // Check if the current item is viewable. Use the item's index as a string for key comparison.
+    const isVisible = viewableItems.has(index.toString());
 
     return (
-      <TouchableOpacity style={styles.videoContainer} onPress={() => {/* Handle play logic if necessary */ }}>
+      <TouchableOpacity
+        style={styles.videoContainer}
+        onPress={() => {
+          setCurrentVideoUrl(videoUrl); // Set the URL for the video to be played
+          setModalVisible(true); // Show the modal
+        }}>
         <Video
-          source={{ uri: item }}
+          source={{ uri: videoUrl }}
           style={styles.video}
           resizeMode="cover"
           controls={true}
-          paused={!isVisible} // Video is paused by default unless it's viewable
-          repeat={true} // Loop video
-          // Important: mute the video by default to comply with auto-play policies
+          paused={!viewableItems.has(index.toString())} // Assume you want to pause when not in view
```

Comment:
does this mean the videos no longer auto-play in the grid?

Reply:
yes! was requiring hefty processing power from the user's device